frontend/app/models/encryptable-credential.js

```diff
@@ -5,13 +5,58 @@ import { isNone } from "@ember/utils";
 export default class EncryptableCredential extends Encryptable {
   @attr("string") cleartextUsername;
   @attr("string") cleartextPassword;
+  @attr("string") cleartextToken;
+  @attr("string") cleartextPin;
+  @attr("string") cleartextEmail;
+  @attr({ defaultValue: undefined }) cleartextCustomAttr;
```

Comment:
We don't really have a data type here, since cleartextCustomAttr is a nested attribute that consists of the two values label and value, we would have a data type defined here, when we add a nested model as a file

  @hasOne("custuom-attr") cleartextCustomAttr;
  
  But in our case we don't really need a nested model, since value and label are just two string fields, which can be defined with the help of methods. Because of that the datatype of cleartextCustomAttr is in our case just a json formatted nested attribute